### PR TITLE
fix: lint fixes based on sonarcloud

### PR DESCRIPTION
**Pull request by IgnatG**
### Changelog
- fixed: Copying recursively might inadvertently add sensitive data to the container.
- fixed: Omitting --ignore-scripts can lead to the execution of shell scripts
- fixed: Promises must be awaited

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,20 +11,20 @@ RUN rm -rf /var/cache/apk/*
...
+COPY --chown=node:node tsconfig*.json .
 
 # Build with all dependencies
 RUN npm run build
 
 # Lean NPM - Only install `dependencies`
 # `devDependencies` will still be resolved inside `package-lock.json`,
 # however they will not be installed inside `node_modules` directory.
-RUN npm ci --legacy-peer-deps --omit=dev
+RUN npm ci --legacy-peer-deps --omit=dev --ignore-scripts
```
Comment (abhi-markan) [inline Dockerfile:27]: I have tried this in the past and it has caused `npm run build` to fail, were you able to build the container without any errors locally?

File: Dockerfile [hunk 2]
```diff
@@ -11,20 +11,20 @@ RUN rm -rf /var/cache/apk/*
...
-COPY --chown=node:node package.json .
-COPY --chown=node:node package-lock.json .
-RUN npm ci --legacy-peer-deps
+COPY --chown=node:node package*.json .
+RUN npm ci --legacy-peer-deps --ignore-scripts
 RUN npm cache clean --force
 
-COPY --chown=node:node . .
+COPY --chown=node:node src src
+COPY --chown=node:node tsconfig*.json .
```
Comment (abhi-markan) [inline Dockerfile:19]: This certainly makes code clean, but should we not be explicit with exact file names?

File: Dockerfile [hunk 3]
```diff
@@ -11,20 +11,20 @@ RUN rm -rf /var/cache/apk/*
...
 # NPM
-COPY --chown=node:node package.json .
-COPY --chown=node:node package-lock.json .
-RUN npm ci --legacy-peer-deps
+COPY --chown=node:node package*.json .
+RUN npm ci --legacy-peer-deps --ignore-scripts
 RUN npm cache clean --force
 
-COPY --chown=node:node . .
+COPY --chown=node:node src src
```
Comment (abhi-markan) [inline Dockerfile:18]: We also need to copy below build dependant file please:

* `nest-cli.json`

File: Dockerfile (hunk 1 above)
Comment (IgnatG) [reply Dockerfile:28]: I've tried it locally and it's working for me. I wonder if it's WSL otherwise?

File: Dockerfile (hunk 2 above)
Comment (IgnatG) [reply Dockerfile:20]: in theory yes. the `*` is just a wild card to copy every file that starts with tsconfig - this is to reduce repetition. I don't mind either way tbh. let me know which one works best for him; happy to update

File: Dockerfile (hunk 3 above)
Comment (IgnatG) [reply Dockerfile:18]: updated now

File: Dockerfile (hunk 1 above)
Comment (IgnatG) [reply Dockerfile]: never mind. it crashed now...
I've removed it

File: Dockerfile (hunk 2 above)
Comment (abhi-markan) [reply Dockerfile:20]: Explicit file names please, thanks

Comment (IgnatG) [reply Dockerfile]: updated now

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**